corehq/apps/reports/templates/reports/tableau_template.html

```diff
@@ -2,18 +2,29 @@
 {% load compress %}
 {% load hq_shared_tags %}
 {% load i18n %}
+{% load static %}
 
 {% block page_content %}
 
 {% initial_page_data "server_type" server_type %}
 {% initial_page_data "ticket" ticket %}
 {% initial_page_data "validate_hostname" validate_hostname %}
+{% initial_page_data "server_address" server_address %}
+{% initial_page_data "target_site" target_site %}
+{% initial_page_data "domain_username" domain_username %}
 {% initial_page_data "view_url" view_url %}
+{% registerurl 'tableau_visualisation_ajax' request.domain %}
 
 <script src="https://{{ validate_hostname }}/javascripts/api/tableau-2.5.0.min.js"></script>
 <script src="{% static 'reports/js/tableau.js' %}"></script>
 
 <body>
+    <div id="loadingDiv">
+        <i class='fa fa-spin fa-spinner' style="font-size: 70px; margin: 50px"></i>
+        {# <img src="{% static "reports/loader.gif" %}" #}
+    </div>
     <div id="vizContainer"></div>
+    <div id="errorMessage"><p>{{ failure_message }}</p></div>
```

Comment:
I think you could give this a class like `alert alert-danger` and that'll give it enough visual distinction that you can then drop the various `p` and `b` tags.